src/klein/test/util.py

```diff
@@ -3,6 +3,9 @@
 """
 
 from abc import ABC, abstractmethod
+from typing import cast
+
+from twisted.trial.unittest import SynchronousTestCase
 
 
 class EqualityTestsMixin(ABC):
```

Comment:
Doing the cast each time you use self looks like unnecessarily extra work :)

I was not able to find a better fix for this.

https://mypy.readthedocs.io/en/stable/class_basics.html#abstract-base-classes-and-multiple-inheritance

Can this just inherit from SynchronousTestCase , instead of ABC ?
And if needed, instead of `abstractmethod` raise a non implemented exception.

I have never used  ABC. So feel free to ignore this.

Reply:
I've tried a few things; mixins are just a pain for typing.